src/lib/traitify-client.js

```diff
@@ -23,6 +23,29 @@ export default class TraitifyClient {
     this.version = version;
     return this;
   }
+  handlePromise = (requestType, xhr, params) => {
+    const async = xhr;
+    const promise = new Promise((resolve, reject) => {
+      if(!this.online()) { return reject(); }
+      try {
+        async.onload = () => {
+          if(async.status === 404) {
+            reject(async.response || async.responseText);
+          } else {
+            resolve(JSON.parse(async.response || async.responseText));
+          }
+        };
+        async.onerror = () => { reject(async.response || async.responseText); };
+        async.ontimeout = () => { reject(async.response || async.responseText); };
+        const send = requestType === "graphql" ? () => { async.send(params); } : () => { xhr.send(JSON.stringify(params)); };
```

Comment:
Also looks like you still reference it as xhr here and below. It'd be fine to leave it as xhr, or change all references